src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -204,6 +204,25 @@ get_leaders_preference(const config_map_t& config) {
     return std::nullopt;
 }
 
+static tristate<std::chrono::milliseconds>
+get_delete_retention_ms(const config_map_t& config) {
+    auto delete_retention_ms = get_tristate_value<std::chrono::milliseconds>(
+      config, topic_property_delete_retention_ms);
+
+    // If the config entry for delete.retention.ms is in the "empty" state, and
+    // the cluster default is also std::nullopt, ensure the option is disabled
+    // by default. DescribeConfigs calls should still return DEFAULT_CONFIG when
+    // describing this state, thanks to override_if_not_default in
+    // config_response_utils.cc.
```

Comment:
Per my other comment, it seems like this seems like a better place to handle things specific to delete retention ms